src/common/imagpng.cpp

```diff
@@ -723,6 +796,29 @@ bool wxPNGHandler::SaveFile( wxImage *image, wxOutputStream& stream, bool verbos
         png_set_pHYs( png_ptr, info_ptr, resX, resY, PNG_RESOLUTION_METER );
 
     png_set_sBIT( png_ptr, info_ptr, &sig_bit );
+
+    // Add custom options
+    const wxArrayString& optionNames = image->GetOptionNames();
+    typedef std::map<std::string, std::string> CustomOptionsMap;
+    CustomOptionsMap customOptions;
+    for (unsigned int optionIndex = 0; optionIndex < optionNames.GetCount(); ++optionIndex)
+    {
+        if (!IsKnownOption(optionNames[optionIndex]))
+            customOptions.insert(CustomOptionsMap::value_type((const char*)optionNames[optionIndex], (const char*)image->GetOption(optionNames[optionIndex])));
+    }
+    info_ptr->num_text = customOptions.size();
+    info_ptr->max_text = info_ptr->num_text * sizeof(png_text);
+    std::auto_ptr<png_text> infoPtrText(new png_text[info_ptr->num_text]); // auto-free when infoPtrText goes out of scope
```

Comment:
For the same reasons as above (but also because `std::auto_ptr` is deprecated in C++17, so using it is already problematic), `wxScopedPtr` should be used here.

There is no need for the comment, C++ programmers  are supposed to know what `auto_ptr` or `scoped pointer` do...